pkg/vsphere/client.go

```diff
@@ -7,37 +7,60 @@ package vsphere
 
 import (
 	"context"
+	"errors"
 	"io/ioutil"
+	"net/http"
 	"net/url"
 	"path/filepath"
+	"time"
 
 	"github.com/kelseyhightower/envconfig"
 	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/session"
+	"github.com/vmware/govmomi/session/keepalive"
 	"github.com/vmware/govmomi/vapi/rest"
+	"github.com/vmware/govmomi/vim25"
+	"github.com/vmware/govmomi/vim25/methods"
 	"github.com/vmware/govmomi/vim25/soap"
+	"knative.dev/pkg/logging"
+
 	corev1 "k8s.io/api/core/v1"
 )
 
 const (
-	VolumeName = "vsphere-binding"
-	MountPath  = "/var/bindings/vsphere" // filepath.Join isn't const.
+	VolumeName        = "vsphere-binding"
+	DefaultMountPath  = "/var/bindings/vsphere" // filepath.Join isn't const.
+	keepaliveInterval = 5 * time.Minute         // vCenter APIs keep-alive
 )
 
 type EnvConfig struct {
-	Insecure bool   `envconfig:"GOVC_INSECURE" default:"false"`
-	Address  string `envconfig:"GOVC_URL" required:"true"`
+	Insecure   bool   `envconfig:"GOVC_INSECURE" default:"false"`
+	Address    string `envconfig:"GOVC_URL" required:"true"`
+	SecretPath string `envconfig:"GOVC_SECRET_PATH" default:""`
 }
 
-// ReadKey may be used to read keys from the secret.
+// ReadKey reads the key from the secret.
 func ReadKey(key string) (string, error) {
-	data, err := ioutil.ReadFile(filepath.Join(MountPath, key))
+	var env EnvConfig
+	if err := envconfig.Process("", &env); err != nil {
+		return "", err
+	}
+
+	mountPath := DefaultMountPath
+	if env.SecretPath != "" {
+		mountPath = env.SecretPath
+	}
+
+	data, err := ioutil.ReadFile(filepath.Join(mountPath, key))
 	if err != nil {
 		return "", err
 	}
 	return string(data), nil
 }
 
-func New(ctx context.Context) (*govmomi.Client, error) {
+// NewSOAPClient returns a vCenter SOAP API client with active keep-alive. Use
+// Logout() to release resources and perform a clean logout from vCenter.
```

Comment:
I wonder if we should be using contextWithCancel and perform these logouts in the keepalive handlers, or some other way if things get cancelled to ensure we get a clean logout?  Not in this PR, but moving forward, seems like we might want to catch signals cleanly and perform the clean logout? 🤔

Reply:
I implemented it in the blocking `Start` method:

```go
func (a *vAdapter) Start(ctx context.Context) error {
	defer func() {
		// using fresh ctx to avoid canceled error during logout
		_ = a.VClient.Logout(context.Background()) // best effort, ignoring error
	}()
...
```

WDYT?

Reply:
I think that's fine for now, I think however that if you run with the client and if we don't catch the signals from the ctx, this might not get a chance to actually run. But we can do that in a follow up, so no need to hold up the PR for that :)